### PR TITLE
fix(shell): canonicalize releases route

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary
- Add canonical `/app/releases` route ownership instead of rewriting it through `/app/dashboard/releases`.
- Reuse the existing releases server/client route contract from both canonical and legacy entry points.
- Keep releases shell loading/error parity and update the rewrite regression tests.

## Verification
- `JOVIE_AGENT_PROFILE=coder pnpm --filter @jovie/web exec vitest run tests/unit/routes/shell-nav-coverage.test.ts tests/unit/routing/chat-route-rewrite.test.ts tests/unit/app/shell-route-matches.test.ts tests/unit/dashboard/dashboard-shell-content.test.ts --config=vitest.config.mts`
- `JOVIE_AGENT_PROFILE=coder pnpm exec biome check 'apps/web/app/app/(shell)/dashboard/releases/page.tsx' 'apps/web/app/app/(shell)/dashboard/releases/ReleasesRoute.tsx' 'apps/web/app/app/(shell)/releases/page.tsx' 'apps/web/app/app/(shell)/releases/loading.tsx' 'apps/web/app/app/(shell)/releases/error.tsx' apps/web/next.config.js apps/web/constants/routes.ts apps/web/tests/unit/routing/chat-route-rewrite.test.ts apps/web/tests/unit/dashboard/dashboard-shell-content.test.ts apps/web/tests/unit/routes/shell-nav-coverage.test.ts`
- `JOVIE_AGENT_PROFILE=coder pnpm --filter @jovie/web run typecheck -- --pretty false`
- `JOVIE_AGENT_PROFILE=coder pnpm turbo typecheck`
- `NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_test_mock NEXT_IGNORE_ESLINT=1 NEXT_IGNORE_TYPECHECK=1 NEXT_PRIVATE_SKIP_SIZE_CHECK=true JOVIE_AGENT_PROFILE=coder pnpm turbo build --filter=@jovie/web`
- `JOVIE_AGENT_PROFILE=coder git diff --check`

## Visual Review
- No production visual rendering delta. This is route ownership and App Router persistence cleanup only; the canonical route renders the same releases client surface and loading/error states.

### CODE REVIEW

**Comment by itstimwhite** [comment]
## Review Summary
**Verdict**: Approve

This PR cleanly canonicalizes the `/app/releases` route by removing the Next.js rewrite and adding a proper App Router page that reuses the existing `ReleasesRoute` component. The legacy `/app/dashboard/releases` entry point is preserved for backward compatibility. All changes are consistent with Jovie conventions, and the verification gates (tests, typecheck, lint) pass.

---

### What Looks Good
- **Shared component extraction** – The server logic is moved into `ReleasesRoute.tsx` and reused from both the canonical and legacy pages, reducing duplication.
- **Loading/error parity** – The new route re‑exports the existing loading and error boundaries, ensuring identical UX.
- **Test coverage** – The rewrite removal is explicitly tested, and the shell data contract test is updated to cover the new canonical route.
- **No visual delta** – The PR correctly claims no production rendering change; the same client components are rendered.
- **Conventional commit** – The PR title follows the required format.

---

### Warnings
None.

---

### Suggestions
- **Consider adding a basic integration test** for the new `/app/releases` page (e.g., a Vitest or Playwright test that verifies the page renders without error). While the existing dashboard tests cover the component, a dedicated route test would guard against future regressions if the route structure changes again.
- **Minor comment clarity** – The updated comment in `constants/routes.ts` is fine, but you could also add a note that `DASHBOARD_RELEASES` is kept only for legacy bookmarks and nested task aliases (already implied). No action required.

---

### Critical Issues
None. The diff is correct, follows Jovie conventions, and passes all verification gates.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).

File: apps/web/app/app/(shell)/dashboard/releases/ReleasesRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * Auth check via getCachedAuth (Clerk JWT, no DB) runs first. The shared shell
+ * and /app route warm the release matrix cache ahead of navigation, so this
+ * page keeps warm transitions instant while still hydrating release data on
+ * direct visits, refreshes, and bookmarks.
+ */
+export async function ReleasesRoute() {
+  const { userId } = await getCachedAuth();
+  if (!userId) {
+    redirect(`${APP_ROUTES.SIGNIN}?redirect_url=${APP_ROUTES.RELEASES}`);
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/app/app/(shell)/dashboard/releases/ReleasesRoute.tsx]: Addressed in `c85554fca` / `f122faebd`: sign-in redirects now use `URLSearchParams`, `/app/releases` is covered as the canonical nav destination, the legacy dashboard releases page is explicitly documented in route coverage, and the new releases error re-export is marked as a client boundary.

**Comment by itstimwhite** [comment]
## Review Summary  
**Verdict**: Changes Requested  

The diff is clean and the overall approach (extracting a shared `ReleasesRoute` server component, removing the rewrite, and adding a canonical `/app/releases` page with loading/error parity) is solid. However, **the new `releases/loading.tsx` re-export pattern is fragile** — it assumes the source file exports a named `ReleaseTableSkeleton`, but this is neither guaranteed nor likely needed. This could break the build silently.

---

### Critical Issues  

1. **`apps/web/app/app/(shell)/releases/loading.tsx` – Unsafe re-export of named export**  
   ```ts
   export { default, ReleaseTableSkeleton } from '../dashboard/releases/loading';
   ```  
   Next.js `loading.tsx` files are **expected to export only a default component**. Re-exporting a named `ReleaseTableSkeleton` from another file is unconventional, and if `../dashboard/releases/loading` does not export a named `ReleaseTableSkeleton`, this will cause a **runtime/build error**.  

   **Fix**: Remove the named export. If you need `ReleaseTableSkeleton` for testing or composability, import it directly where needed, or keep it as a re‑export only if the source `loading.tsx` explicitly exports it. Better yet, create a simple default skeleton wrapper in the canonical route file rather than relying on an alias.  
   ```ts
   // apps/web/app/app/(shell)/releases/loading.tsx
   export { default } from '../dashboard/releases/loading';
   ```

---

### Warnings  

2. **`apps/web/app/app/(shell)/releases/error.tsx` – Same re-export pattern but less risky**  
   ```ts
   export { default } from '../dashboard/releases/error';
   ```  
   This is fine for a `error.tsx` as long as the source exports a default `'use client'` component. However, ensure the source file actually exists and is a client component – missing imports here would also break the build. Consider adding a brief sanity test to verify that both the canonical and legacy error boundaries render without crashes.

---

### Suggestions  

3. **Extract shared loading/error into a reusable module**  
   Both `/app/releases` and `/app/dashboard/releases` now share the same error and loading implementations via re-exports. While this works, a cleaner approach would be to place the shared `loading.tsx` and `error.tsx` files inside a common folder (e.g., `apps/web/app/app/(shell)/_shared/`) and let both route directories import from there. This avoids fragile cross‑directory re-exports and makes the dependency explicit.  

4. **Test that both routes render identical content**  
   The PR updates shell‑route‑matching tests and the dashboard‑shell‑data test, but does not add a test asserting that visiting `/app/releases` and `/app/dashboard/releases` serve the same UI (e.g., via a Playwright screenshot test). Consider adding one to prevent future divergence.

5. **Minor comment improvement in `routes.ts`**  
   The updated comment is good, but consider adding a `@deprecated` JSDoc tag to `DASHBOARD_RELEASES` to signal to future developers that new code should use `RELEASES` instead.

---

### What Looks Good  

- **Clean extraction** of the server logic into `ReleasesRoute.tsx` – single source of truth, reused from both route entry points.  
- **Removal of the Next.js rewrite** – reduces server‑side routing overhead and makes the route ownership explicit.  
- **Updated convention comment** in `routes.ts` clarifies the legacy route’s purpose.  
- **Test updates** accurately reflect the new route structure:  
  - `shouldUseEssentialShellData` now covers both paths.  
  - Shell‑nav‑coverage test removes the stale alias map.  
  - Chat‑rewrite test confirms the release rewrite is gone.  
- **Preservation of loading/error boundaries** ensures no visual regression during transitions.  
- **Conventional commit message** and thorough verification steps.

---

**Action required**: Fix the `loading.tsx` re-export before merging. Everything else is good to ship after that change.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).
